File: tests/fixtures/sample.ts
/**
 * Sample TypeScript module for testing
 */

/**
 * Formats a name with a title
 */
export function formatName(first: string, last: string): string {
    return `${first} ${last}`;
}

/**
 * Calculates the average of numbers
 */
export function average(numbers: number[]): number {
    if (numbers.length === 0) return 0;
    return numbers.reduce((a, b) => a + b, 0) / numbers.length;
}

/**
 * Arrow function to double a number
 */
export const double = (x: number): number => x * 2;

/**
 * A simple user class
 */
export class User {
    constructor(public name: string, public age: number) {}

    /**
     * Gets a greeting for this user
     */
    greet(): string {
        return `Hello, I'm ${this.name}`;
    }

    /**
     * Checks if user is adult
     */
    isAdult(): boolean {
        return this.age >= 18;
    }
}

File: tests/fixtures/sample.py
"""Sample Python module for testing."""

def greet(name: str) -> str:
    """Return a greeting for the given name."""
    return f"Hello, {name}!"

def calculate_sum(numbers: list[int]) -> int:
    """Calculate the sum of a list of numbers."""
    return sum(numbers)

class Counter:
    """A simple counter class."""

    def __init__(self, start: int = 0):
        """Initialize counter with a starting value."""
        self.value = start

    def increment(self):
        """Increment the counter by 1."""
        self.value += 1

    def decrement(self):
        """Decrement the counter by 1."""
        self.value -= 1

    def get(self) -> int:
        """Get the current counter value."""
        return self.value
